src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH_OF_PASSWORD_AND_LOGIN = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("The user with the same login already exists");
+        }
+
+        if (storageDao.get(user.getLogin()) == null) {
+            if (user.getLogin() == null) {
+                throw new InvalidDataException("The login cannot be null");
+            }
+            if (user.getPassword() == null) {
+                throw new InvalidDataException("The password cannot be null");
+            }
+            if (String.valueOf(user.getAge()) == null) {
+                throw new InvalidDataException("The age cannot be null");
```

Comment:
Using `String.valueOf(user.getAge()) == null` is incorrect for checking if age is null because `getAge()` returns a primitive int, which can never be null. Instead, you should check if `user.getAge()` is a valid value (e.g., non-negative).